src/rgw/driver/rados/rgw_sal_rados.cc

```diff
@@ -2421,7 +2421,7 @@ int RadosMultipartUpload::complete(const DoutPrefixProvider *dpp,
   int marker = 0;
   uint64_t min_part_size = cct->_conf->rgw_multipart_min_part_size;
   auto etags_iter = part_etags.begin();
-  rgw::sal::Attrs attrs = target_obj->get_attrs();
+  rgw::sal::Attrs& attrs = target_obj->get_attrs();
```

Comment:
@cbodley can you please see if this change has other implication?

Reply:
it's hard to tell. are we mutating the attrs? if not these should be const-refs instead

Reply:
yes we are mutating the attrs at [line 2544](https://github.com/ceph/ceph/pull/54569/files#diff-8d5ab77cf6b24e95c2c7b352c52d1e254781f6c9e7fb3fd148bddc8a3f4321e7R2544):
```
  attrs[RGW_ATTR_ETAG] = etag_bl;

  if (compressed) {
    // write compression attribute to full object
    bufferlist tmp;
    encode(cs_info, tmp);
    attrs[RGW_ATTR_COMPRESSION] = tmp;
  }
```